### PR TITLE
Add support of Traffic-routing spec

### LINKED ISSUE #16: Add support of Traffic-routing spec
OpenSergo need a new configuration type to do label routing

### PR DESCRIPTION
**Pull request by 123liuziming**
### Does this pull request fix one issue?

refrence to issue [16](https://github.com/opensergo/opensergo-java-sdk/issues/16)

### Describe how you did it
Add xds proto so that OpenSergo can listen the route configuration
### Special notes for reviews

### CODE REVIEW

File: src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java [hunk 1]
```diff
@@ -42,6 +43,7 @@ public final class OpenSergoConfigKindRegistry {
         registerConfigKind(ConfigKind.THROTTLING_STRATEGY, ThrottlingStrategy.class);
         registerConfigKind(ConfigKind.CONCURRENCY_LIMIT_STRATEGY, ConcurrencyLimitStrategy.class);
         registerConfigKind(ConfigKind.CIRCUIT_BREAKER_STRATEGY, CircuitBreakerStrategy.class);
+        registerConfigKind(ConfigKind.TRAFFIC_ROUTER_STRATEGY, RouteConfiguration.class);
```
Comment (jnan806) [inline src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java:46]: `RouteConfiguration.class` should refers to `opensergo`,  but the import Class is `io.envoyproxy.envoy.config.route.v3.RouteConfiguration`

**Review by jnan806 — COMMENTED** [review]
and the `traffic-routing` proto files of `opensergo` should be added into directory `src/main/proto/`

File: src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java (hunk 1 above)
Comment (123liuziming) [reply src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java:46]: We didn't change envoy's RouteConfiguration at all, so we just reuse the RouteConfiguration provided by envoy

**Comment by 123liuziming** [comment]
> and the `traffic-routing` proto files of `opensergo` should be added into directory `src/main/proto/`
Therer is no traffic-routing proto, we just reuse the proto in xds

**Comment by sczyh30** [comment]
> and the `traffic-routing` proto files of `opensergo` should be added into directory `src/main/proto/`

Actually we just leveraged the original "route configuration" from xDS RDS proto (and there are some extension fields inside it, which may satisfy our requirement)

**Comment by sczyh30** [comment]
cc @123liuziming @panxiaojun233 We may need to add some description for our design later.

File: src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java (hunk 1 above)
Comment (sczyh30) [inline src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java:46]: We may need to add a comment for this (we've reused the route proto from xDS proto).

**Review by sczyh30 — APPROVED** [review]
LGTM

File: src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java (hunk 1 above)
Comment (sczyh30) [reply src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java:46]: @jnan806 Any other suggestions?

Comment (jnan806) [reply src/main/java/io/opensergo/OpenSergoConfigKindRegistry.java:46]: no more ~  😃

**Review by jnan806 — APPROVED** [review]
LGTM

**Comment by jnan806** [comment]
cc @sczyh30 @123liuziming @panxiaojun233 I will merge it with `squash and merge`

**Comment by sczyh30** [comment]
Nice work. Thanks!

**Pull request merged**